src/govuk/components/skip-link/skip-link.js

```diff
@@ -0,0 +1,81 @@
+function SkipLink ($module) {
+  this.$module = $module
+}
+
+/**
+ * Initialise the component
+ */
+SkipLink.prototype.init = function () {
+  var $module = this.$module
+  // Check for module
+  if (!$module) {
+    return
+  }
+
+  $module.addEventListener('click', this.handleClick.bind(this))
+}
+
+/**
+* Click event handler
+*
+* @param {MouseEvent} event - Click event
+*/
+SkipLink.prototype.handleClick = function (event) {
+  var target = event.target
+
+  if (this.focusTarget(target)) {
+    event.preventDefault()
+  }
+}
+
+/**
+ * Focus the target element
+ *
+ * @param {HTMLElement} $target - Event target
+ * @returns {boolean} True if the target was able to be focussed
+ */
+SkipLink.prototype.focusTarget = function ($target) {
+  // If the element that was clicked does not have a href, return early
+  if ($target.href === false) {
+    return false
+  }
+
+  var contentId = this.getFragmentFromUrl($target.href)
+  var $content = document.getElementById(contentId)
+  if (!$content) {
+    return false
+  }
+
+  if (!$content.getAttribute('tabindex')) {
+    // Set the content tabindex to -1 so it can be focused with JavaScript.
+    $content.setAttribute('tabindex', '-1')
+    $content.classList.add('govuk-skip-link-focused-element')
+
+    // Remove the tabindex attribute on blur because the content only needs to be focusable until it
+    // has received programmatic focus and a screen reader has announced it.
+    $content.addEventListener('blur', function () {
+      $content.removeAttribute('tabindex')
+      $content.classList.remove('govuk-skip-link-focused-element')
+    })
```

Comment:
I don't think this is true, it's valid to use an anonymous function with an event listener: https://developer.mozilla.org/en-US/docs/Web/API/EventListener#javascript I don't think a named function would change the actual behaviour in any way. But I'm going to leave this open, we can compare notes on Monday 👀

Reply:
My bad, I've had another look and this was actually a good catch 💯  I'm now binding the function on the event listener which stops the event listener being created more than once.

Reply:
This still isn't working as expected - I want to refactor the code so that we set the event listener at the point of initialisation but going to pick this up later so that we can get the actual change to the component behaviour tested as part of the pre-release.